pydantic_settings/sources.py

```diff
@@ -602,7 +602,9 @@ def read_env_file(
         return file_vars
 
 
-def _annotation_is_complex(annotation: type[Any] | None) -> bool:
+def _annotation_is_complex(annotation: type[Any] | None, metadata: list[Any]) -> bool:
+    if any(str(md) == 'Json' for md in metadata):
```

Comment:
Not sure I am doing the right way to check if the field is a `Json` field.

Reply:
Yeah this seems to be right:
```suggestion
    if any(isinstance(md, Json) for md in metadata):
```

(Also need to add `Json` to the set of imports from `pydantic` at the top)

Reply:
Thanks for your suggestion. Done